### PR TITLE
ID-3032 [FIX] Fixed the settings for the date range default value 

### PR DESCRIPTION
**Pull request by armine-fliplet**

### CODE REVIEW

File: js/components/dateRange.js [hunk 1]
```diff
@@ -188,48 +208,68 @@ Fliplet.FormBuilder.field('dateRange', {
...
       }
     },
-    formatLocaleDate: function(date) {
-      return date.locale('en').format('YYYY-MM-DD');
+    formatDate: function(date) {
+      return date ? moment(date).locale('en').format('YYYY-MM-DD') : moment(date).locale('en').format('YYYY-MM-DD');
+    },
+    validateValue: function(value) {
+      if (!value || (!value.start && !value.end)) {
+        value = {
```
Comment (tonytlwu) [inline js/components/dateRange.js:256]:
```suggestion
        value = null;
```

File: js/components/dateRange.js [hunk 2]
```diff
@@ -188,48 +208,68 @@ Fliplet.FormBuilder.field('dateRange', {
...
+          start: '',
+          end: ''
+        };
+      } else if (typeof value !== 'object') {
+        value = {
+          start: moment(value, 'YYYY-MM-DD', true).isValid() ? this.formatDate(value) : this.formatDate(),
```
Comment (tonytlwu) [inline js/components/dateRange.js:262]:
```suggestion
          start: this.formatDate(value),
          end: this.formatDate(value)
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/components/dateRange.js [hunk 3]
```diff
@@ -188,48 +208,68 @@ Fliplet.FormBuilder.field('dateRange', {
...
+          end: moment(value, 'YYYY-MM-DD', true).isValid() ? this.formatDate(value) : this.formatDate()
+        };
+      } else if (!value.start || !value.end) {
+        value = {
+          start: value.start || value.end,
+          end: value.end || value.start
+        };
+      }
```
Comment (tonytlwu) [inline js/components/dateRange.js:270]:
```suggestion
      } else if (value.start > value.end) {
        value.end = value.start;
      }
```

File: js/components/dateRange.js [hunk 4]
```diff
@@ -4,18 +4,33 @@ Fliplet.FormBuilder.field('dateRange', {
   props: {
     value: {
       type: Object,
-      default: null
+      default: {
```
[the remark this answers is not in the document]
Comment (armine-fliplet) [reply js/components/dateRange.js:7]: I was getting an error in `set` function, but it's from the API, it's okay now. Fixed!

File: templates/configurations/dateRange.interface.hbs [hunk 5]
```diff
@@ -38,6 +44,22 @@
   </div>
 </div>
 
+<div v-else class="form-group">
```
[the remark this answers is not in the document]
Comment (armine-fliplet) [reply templates/configurations/dateRange.interface.hbs:47]: I had a task in the current phase, and in the comments Sofi and you have left a comment about adding 3rd option.

**Review by tonytlwu — APPROVED** [review]

**Pull request merged**